### DISCUSSION TITLE
PDF merge incorrect order

### DISCUSSION THREAD

**Discussion by 11laflame** [gotenberg/gotenberg#701] [Q&A] [answered]
Hi, I'm using version 7 and currently have a problem that output is not a same order that I delivered

I'm using request like in your documentation

curl \
--request POST 'http://localhost:3000/forms/libreoffice/convert' \
--form 'files=@"/path/to/file.docx"' \
--form 'files=@"/path/to/file.xlsx"' \
--form 'merge="true"' \

**Comment by gulien** [comment]
I have to update the documentation, but the documents are merged in an alphabetical manner.

**Reply by 11laflame** [reply]
And If I will send buffer instead of path to file, on which statement order will be?
It seems like not alphabetical

Example of my formdata

FormData {
  _overheadLength: 619,
  _valueLength: 1731245,
  _valuesToMeasure: [],
  writable: false,
  readable: true,
  dataSize: 0,
  maxDataSize: 2097152,
  pauseStreams: true,
  _released: false,
  _streams: [
    '----------------------------694197593913063554830909\r\n' +
      'Content-Disposition: form-data; name="files"; filename="pdfa.pdf"\r\n' +
      'Content-Type: application/pdf\r\n' +
      '\r\n',
    <Buffer 25 50 44 46 2d 31 2e 34 0a 25 c3 a4 c3 bc c3 b6 c3 9f 0a 32 20 30 20 6f 62 6a 0a 3c 3c 2f 4c 65 6e 67 74 68 20 33 20 30 20 52 2f 46 69 6c 74 65 72 2f ... 1692791 more bytes>,
    [Function: bound ],
    '----------------------------694197593913063554830909\r\n' +
      'Content-Disposition: form-data; name="files"; filename="Screenshot from 2023-10-06 13-29-48.png"\r\n' +
      'Content-Type: image/png\r\n' +
      '\r\n',
    <Buffer 89 50 4e 47 0d 0a 1a 0a 00 00 00 0d 49 48 44 52 00 00 02 6f 00 00 01 5c 08 06 00 00 00 9e 30 59 16 00 00 00 04 73 42 49 54 08 08 08 08 7c 08 64 88 00 ... 22920 more bytes>,
    [Function: bound ],

**Reply by gulien** [reply, marked as answer]
Still, a form file has a filename associated to it, it does not care how you create the multipart/form-data request.

**Reply by 11laflame** [reply]
Yeap, thanks a lot

**Comment by signalwerk** [comment]
@gulien I just tried for the first time the merge function. and was also wondering why the files are sorted first. May I ask you what the reasoning behind it is? It feels counterintuitive to have a pdf-merge but not be able to post the files in an order but need to maintain the filename. I wonder what the use case for merge is thought for?

**Reply by gulien** [reply]
Mainly because I don’t know the order of the files in Gotenberg. All I have is a bunch of files that need to be merged, so the easiest way is to use their names to determine the order 😅

**Reply by gulien** [reply]
I’m not sure I can trust the order here: https://github.com/gotenberg/gotenberg/blob/b8926350b9ea4a7cc624bde5a97350c9128a677f/pkg/modules/api/context.go#L164

**Reply by signalwerk** [reply]
@gulien Thank you for your response. I like the pragmatic way 🤓. I renamed now my files, but it's an extra step that feels a bit wrong. I don't know the language go, but I'd be surprised if it's not in order of the form sent. But hey; always open for a surprise 🫣
